libraries/stdlib/stdlib_ng.mtlx

```diff
@@ -1071,6 +1073,36 @@
     <output name="out" type="vector2" nodename="N_addpivot" />
   </nodegraph>
 
+  <!--
+    Node: <place2dframe>
+    Transform incoming UV texture coordinates from one 2D frame of reference to another.
+    The order of transformation is: -pivot, translate, rotate, scale, pivot.
+  -->
+  <implementation name="IMPL_place2dframe_vector2" nodedef="ND_place2dframe_vector2" nodegraph="NG_place2dframe_vector2" />
```

Comment:
Although this new implementation-based pattern for assigning nodegraphs to nodedefs is supported in 1.38, I wonder if it would be better to align with the approach used by other graph definitions in this file.  Notice that in nodes such as "contrast" below, we're simply assigning the nodegraph to its nodedef with a single attribute, and no intermediate implementation element is required.

Reply:
I switched to this implementation pattern as suggested by @kwokcb, https://github.com/AcademySoftwareFoundation/MaterialX/pull/1027#discussion_r919596826
Do you mean that it's better to revert my latest change for assigning nodegraph NG_place2d_vector2 and NG_place2dframe_vector2 to their nodedefs in [e356b76](https://github.com/AcademySoftwareFoundation/MaterialX/commit/e356b7674aa7a413daf1b6daf0036235c6a67341)?

Reply:
If we don't need implementation node It seems we don't need the node definition `ND_place2dframe_vector2`, just use original `ND_place2d_vector2` for 2 node graphs as below: `<nodegraph name="NG_place2dframe_vector2" nodedef="ND_place2d_vector2">`

Reply:
@vernalchen Yes, that's the change I would recommend, where you associate the nodegraphs with their nodedefs directly through attributes on the nodegraph elements:

```
<nodegraph name="NG_place2dframe_vector2" nodedef="ND_place2dframe_vector2">
```

@francis-wangfr I believe we'll still need two separate nodedef elements for `place2d` and `place2dframe`, so that they are declared as two separate nodes in the library.  Let me know, though, if I'm missing a detail in this regard.

Reply:
That sounds like a good suggestion to me, @dbsmythe, and it would eliminate the need for two distinct nodes with very similar behaviors in common cases.

Reply:
I will mock up a suggested update to the node description in the v1.39 Supplemental Notes document and we can hash out details there.

Reply:
Here is the initial description for the added input, feel free to suggest alternate wording:
`operationorder` (integer enum): the order in which to perform the transform operations. "0" or "SRT" performs -pivot scale rotate translate +pivot as per the original implementation matching the behavior of certain DCC packages, and "1" or "TRS" performs -pivot translate rotate scale +pivot which does not introduce texture shear.  Default is 0 "SRT" for backward compatibility.

Reply:
I will add my 2-cents for counter arguments :).
- There is already an existing 1:1 mapping to other APIs like USD which assume one behaviour and now there this node adds new possible behaviour based on a toggle. Basically overloading the existing behaviour so downstream implementations may need to add additional branching logic. 
- This adds another case were a "value change" can result in the requirement for new code generation. If done at the graph level we need to introduce an if statement which is also not good for performance as both paths needs to be emitted.
- @jstone-lucasfilm , as for using implementation, I suggested this to allow migration to the "new" way of association.
I would assume the nodedef inside a nodegraph would be deprecated.

Reply:
@kwokcb For the usage of an implementation element, though, wouldn't we want to make this change globally in our data libraries, rather than piecemeal as new node definitions are added?  I can't see an advantage in creating a mixture of both techniques in the data libraries, when we're currently using one approach consistently.

Reply:
Re: @kwokcb 's concerns:
1) I think that even with this proposal, there's still a 1:1 mapping, it's just with the new input value hard-coded to "0" (or "1", depending on which behavior you want to match).
2) I think that in the *vast* majority of cases, the `operationorder` input would be set to a static value at the time the node is generated or certainly by the time initial code generation is done.  And in this situation, I'd expect the code generation to include both branches if the "if" or "switch", but one side of the branch would be optimized out due to realizing the `operationorder` value is constant.
3) Agreed with @jstone-lucasfilm that any switch to using implementations for nodedef-nodegraph connections would be done all at once, *if* we agree that this is a desirable thing to do.  (it's one thing to say this is supported, and a different thing to say it's preferred.)

Reply:
So we have reached an agreement that we should add an `operationorder` input to` place2d` to switch between two behaviors, right?

Reply:
Yes, @vernalchen, it looks we need a enum switch input `operationorder ` in the place2d node that has default value as original order SRT (0), and we set different order RTS(1) to get what we want. Thanks for your guide and discussion, @kwokcb @jstone-lucasfilm @dbsmythe , let us know if there is not the final solution.

Reply:
Hi @jstone-lucasfilm @dbsmythe @ashwinbhat @francis-wangfr , please help review the latest changed committed recently, thanks!